### DISCUSSION TITLE
Logging Timezone (TZ)

### DISCUSSION THREAD

**Discussion by raegedoc** [simple-login/app#1410] [Feature Requests]
Hi !  For debugging purposes, it's easier to have time in localtime vs gmtime.

I propose to change _get_console_handler() in app/log.py to replace gmtime with localtime:

```
def _get_console_handler():
    console_handler = logging.StreamHandler(sys.stdout)
    console_handler.setFormatter(_log_formatter)
    # console_handler.formatter.converter = time.gmtime
    console_handler.formatter.converter = time.localtime

    return console_handler
```

Now, to get logs in localtime, all is required to do is setting TZ environment variable to local timezone.  The complete list of TZ can be found on Wikipedia (https://en.wikipedia.org/wiki/List_of_tz_database_time_zones)

For exemple, in New-York (GMT-5) we would specify TZ=America/New_York:

```
docker run -d \
    --name sl-app \
    -e TZ=America/New_York \
    -v $(pwd)/sl:/sl \
    -v $(pwd)/sl/upload:/code/static/upload \
    -v $(pwd)/simplelogin.env:/code/.env \
    -v $(pwd)/dkim.key:/dkim.key \
    -v $(pwd)/dkim.pub.key:/dkim.pub.key \
    -p 127.0.0.1:7777:7777 \
    --restart always \
    --network="sl-network" \
    simplelogin/app:3.4.0
```

And in docker-compose format, it would look like:

```
services:
  sl-app:
    image: simplelogin/app:3.4.0
    ...
    environment:
      - TZ=America/New_York
    ...
```

If TZ is not specified, TZ=Etc/GMT and SL logging is as usual.

What do you think about that ?

**Comment by nguyenkims** [comment]
@Raegedoc as a convention, server log actually always uses UTC. This is handy if people in a team live across different timezones.

**Comment by raegedoc** [comment]
@nguyenkims Having the option is very useful when self-hosting.  Plus, if TZ is not specified it still set to UTC and teams friendly across different timezones.

**Reply by nguyenkims** [reply]
I see, in this case feel free to create a pull request to add that feature.
